app/controllers/api/restful_controller.rb

```diff
@@ -18,7 +20,8 @@ def destroy_action
   private
 
   def current_user
-    super || LoggedOutUser.new
+    token_user = User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
```

Comment:
Also, I reckon caching the doorkeeper user so we don't have to hit the database every time we reference the current user would be nice.

```
def current_user
  super || token_user || LoggedOutUser.new
end

private

def token_user
  @token_user ||= User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
end
```